doc/source/programs/gdal_dataset_identify.rst

```diff
@@ -27,10 +27,22 @@ Options
 
     Any file name or directory name. Required. May be repeated
 
-.. option:: -f, --of, --format, --output-format json|text
+.. option:: -o, --output <OUTPUT>
+
+    .. versionadded:: 3.13
+
+    Output filename. Not needed for JSON or text output where standard output
```

Comment:
It will certainly become clearer as I proceed in the review but at this point I haven't yet fully understood how the output of this command will be stored in the output filename. Is it added to metadata?
Can you please clarify how the information is stored in the <OUTPUT>?
Also, you start with listing the exceptions to when the argument is mandatory, I would find it more natural to start saying that the argument is required and list the exceptions last (unless you believe that the majority of use cases will not make it mandatory).